### DISCUSSION TITLE
What are the recommended hardware requirements?

### DISCUSSION THREAD

**Discussion by teuchezh** [TabbyML/tabby#2709] [Q&A]
What hardware is suitable for comfortable work? Are you interested in the processor model, the amount of RAM, what is the recommended video card if you use it?
I’ll assume that you need at least an Intel processor with OpenVINO support, for example, on the secondary market we have a GTX1060 P106-100, on average $30-35 with 6GB video memory, how suitable is this option?
Is Intel Compute Stick 2 supported?

**Comment by mawisec** [comment]
I would also like to know a recommended PC setup.

**Comment by Cfomodz** [comment]
### CPU-Only Setup
- **Processor**: Modern multi-core CPU (Intel or AMD)
- **RAM**: Minimum 8GB, **16GB recommended**
- **Storage**: At least 10GB free space for framework and models
- **Operating System**: 
  - Linux (Ubuntu 18.04+ recommended)
  - macOS
  - Windows (via WSL or Docker)

### GPU-Accelerated Setup (Recommended)
Running Tabby with GPU acceleration significantly improves performance and is highly recommended for production use.

---

### Small Models (1B-3B Parameters)

**Recommended GPUs:**
- NVIDIA T4 (16GB VRAM)
- NVIDIA GTX 10 Series (1060 6GB or higher)
- NVIDIA GTX 20 Series
- Apple Silicon (M1/M2/M3/M4)

**VRAM Requirements:**
- **1B models**: ~2-4GB VRAM
- **3B models**: ~4-6GB VRAM

**Example Models:**
- StarCoder-1B
- StarCoder2-3B
- DeepSeekCoder-1.3B
- Qwen2.5-Coder-1.5B
